provider/bidengine/order.go

```diff
@@ -82,6 +126,16 @@ func newOrderInternal(svc *service, oid mtypes.OrderID, cfg Config, pass Provide
 
 var matchBidNotFound = regexp.MustCompile("^.+bid not found.+$")
 
+func observeRunner(f func () runner.Result, observer prometheus.Histogram) func() runner.Result{
```

Comment:
kinda feel like this can be added to a `runner.Do`-ish variant but we can... _do_ that later if we find that we want this elsewhere.